thumbor.js

```diff
@@ -20,6 +20,58 @@ class CpmThumbor{
         this.THUMBOR_RESIZE_DOMAINE_PREFIX = resizeDomainPrefix;
     }
 
+    getFormat (format) {
+        let back = {};
+        switch (format) {
+            case 'xs':
+                back = { width: 100, height: 75, cropMode: "smart"};
+                break;
+            case 'md':
+                back = { width: 200, height: 200, cropMode: "fitIn"};
+                break;
+            case 'lg':
+                back = { width: 400, height: 400, cropMode: "fitIn"};
+                break;
+            case 'lgcrop':
+                back = { width: 400, height: 300, cropMode: "smart"};
+                break;
+            case 'legacy_vignette':
+                back = { width: 150, height: 113, cropMode: "fitIn"};
+                break;
+            case 'legacy_picture':
+                back = { width: 640, height: 480, cropMode: "fitIn"};
+                break;
+            case 'legacy_jsonhd':
+            case 'pswp':
+                back = { width: 1920, height: 1080, cropMode: "fullFitIn"};
+                break;
+            case 'mdViewGallery':
+                back = { width: 176, height: 132, cropMode: "fullFitIn"};
+                break;
+            case 'original':
+                back = {};
+                break;
+            case 'sm':
+            default:
+                back = { width: 160, height: 120, cropMode: "smart"};
+                break;
+        };
+        return back;
+    }
+
+    generateRewriteImgUrl(mediaUrl, rewriteParameters) {
+        let parametersXs = Object.assign({ ... rewriteParameters }, this.getFormat("xs"));
+        let parametersMd = Object.assign({ ... rewriteParameters }, this.getFormat("md"));
+        let parametersLg = Object.assign({ ... rewriteParameters }, this.getFormat("lg"));
```

Comment:
Je ne vois pas ce qui change par rapport à : 
```js
let parametersLg = Object.assign(rewriteParameters, this.getFormat("lg"));
```

Reply:
Le probleme est que object.assign modifie le premier parametre pour y ajouter les infos des objets suivant.
J'utilise l'operateur de destructuration ES6 pour faire une copie du parametre:
`https://codeburst.io/a-simple-guide-to-destructuring-and-es6-spread-operator-e02212af5831`

Reply:
Vu ensemble, c'est un shortcut pour faire des copies de l'objet.
J'y ai réfléchi un peu et je trouve un peu plus clair l'écriture suivante : 
```js
let parametersLg = Object.assign({}, rewriteParameters, this.getFormat("lg"));
```